lib/storage/metadata/mongoclient/readStream.js

```diff
@@ -60,12 +60,21 @@ class MongoReadStream extends Readable {
         }
 
         if (searchOptions) {
-            Object.assign(query, searchOptions);
+            this._cursor = c.aggregate([
+                { $match: searchOptions }, // user query
+                { $match: query }, // for paging
```

Comment:
was wondering if the pipeline would benefit from first filter by the indexed `_id`, then followed by the `{ $match: searchOptions}`, which can possibly involve a full collection scan

Reply:
What we want is first to do the user defined query: searchOptions, then save it on disk (potentially larger results set than 100MB), then sort this results set.

Reply:
But I think we need to add a `$out` stage into a temporary collection and obtain a cursor on this temp collection instead.
Later on a sweeper shall delete temp collections.